exercises/resistor-color-trio/canonical-data.json

```diff
@@ -0,0 +1,54 @@
+{
+  "exercise": "resistor-color-trio",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Orange and orange and black",
+      "property": "value",
+      "input": {
+        "colors": ["orange", "orange", "black"]
+      },
+      "expected": 33
+    },
+    {
+      "description": "Green and brown and brown",
+      "property": "value",
+      "input": {
+        "colors": ["green", "brown", "orange"]
+      },
+      "expected": 510
+    },
+    {
+      "description": "Blue and grey and yellow",
+      "property": "value",
+      "input": {
+        "colors": ["blue", "grey", "yellow"]
+      },
+      "expected": 680000
+    },
+    {
+      "description": "Brown and black and black",
+      "property": "description",
+      "input": {
+        "colors": ["brown", "black", "black"]
```

Comment:
The inputs for the `description` property should be integers, not color lists. The student has already written a three-color conversion function, at this point. No need to force them to call it; it is both obvious and natural to hook the output of the color converter into the input of the describer.

Reply:
Except that the integers don't _mean_ anything by themselves. 

"orange" "orange" "orange" would be [3, 3, 3].

```
(10 * colorValue('orange') + colorValue('orange')) * (10 ** colorValue('orange'))
```

Would be a readable solution. If we use integers, we are moving away from correct and optimal solutions that look like this:

```
(colorFirstValue('orange') + colorSecondValue('orange')) * colorMultiplier('orange')
```

Reply:
I didn't mean to suggest a list of integers; I meant a single integer, the value in ohms.

The intent is to get people to write one `calculate_value` function, and one `format_description` function. It should be obvious and trivial to nest the calls: if what you have is a list of colors, and what you want is a string description, there's no reason not to write `format_description(calculate_value(color_list))`. 

However, it would teach bad factorization to include both operations in a single function.

Reply:
Ah. I understand what you're (both, @coriolinus and @yawpitch) are saying. I think having `value` return 3300 is fine 👍 😄 .

I think having `label` return `"33 kilo Ohms"` is also something we should have.
I think having `description` is something a track can add based on preference on a sentence.

I don't agree with all the "this will lead to bad code" arguments. I _understand_ what you're saying, but it's up to the student to recognise the two concerns or for the mentor to point them out. Purposefully crafting problem descriptions so a student can't do it wrongly is kinda forcing a certain implementation, which is something we don't want.

Reply:
See: https://github.com/exercism/problem-specifications/pull/1551#issuecomment-512589784

`value` as number (33000) is still an incorrect abstraction.

Reply:
> I personally think that _forcing description to take "ohms"_ is also a language-specific implementation details. Having a class object with a description _property_ that doesn't take arguments is very reasonable in a lot of languages. Both your arguments are forcing a certain code style (that is functional).

I'm not sure that it does; it's up to the track implementer to decide on function vs class as the organizing structure, and for instance in Python I might reasonably decide that this should be the excercise in which we introduce subclassing of immutables and implement this as:

```python
class ResistorValue(int):
    def __new__(cls, color, *colors):
         pass

    @property 
    def description(self):
        pass
```

Now `description` is perfectly valid and doesn't take ohms as a functional parameter, because the class itself is inherently a value in ohms.